cluster/gce/util.sh

```diff
@@ -771,39 +771,78 @@ function create-nodes() {
 # - NODE_INSTANCE_PREFIX
 # - PROJECT
 # - ZONE
+# - AUTOSCALER_MAX_NODES
+# - AUTOSCALER_MIN_NODES
+function create-cluster-autoscaler-mig-config() {
+
+  # Each MIG must have at least one node.
+  if $((${AUTOSCALER_MIN_NODES} < ${NUM_MIGS})); then
+    echo "AUTOSCALER_MIN_NODES must be greater or equal ${NUM_MIGS}"
+    exit 2    
+  fi
+
+  # Each MIG must have at least one node.
+  if $((${AUTOSCALER_MAX_NODES} < ${NUM_MIGS})); then
+    echo "AUTOSCALER_MAX_NODES must be greater or equal ${NUM_MIGS}"
+    exit 2    
+  fi
+
+  local min_nodes_left=$((${AUTOSCALER_MIN_NODES}-${NUM_MIGS}))
+  local max_nodes_left=$((${AUTOSCALER_MAX_NODES}-${NUM_MIGS}))
+  local instances_per_mig=$(((${NUM_NODES} + ${NUM_MIGS} - 1) / ${NUM_MIGS}))
+  local last_mig_size=$((${NUM_NODES} - (${NUM_MIGS} - 1) * ${instances_per_mig}))
+  
+  AUTOSCALER_MIG_CONFIG=""
+
+  for ((i=1; i<${NUM_MIGS}; i++)); do
+    local mig_url=`gcloud compute instance-groups describe
+      "${NODE_INSTANCE_PREFIX}-group-$i" 
+      --zone "${ZONE}"
+      --project "${PROJECT}" | grep selfLink | awk '{print $2}'`
+    
+    local this_min=$(($instances_per_mig>$min_nodes_left?$min_nodes_left:$instances_per_mig))
+    local this_max=$(($instances_per_mig>$max_nodes_left?$max_nodes_left:$instances_per_mig))
+
+    min_nodes_left-=${this_min}
+    max_nodes_left-=${this_max}
+
+    local mig_min=$((${this_min}+1))
+    local mig_max=$((${this_max}+1))
+
+    AUTOSCALER_MIG_CONFIG="${AUTOSCALER_MIG_CONFIG} --nodes=${mig_min}:${mig_max}:${mig_url}"
+  done
+
+  local mig_url=`gcloud compute instance-groups describe 
```

Comment:
Please add a comment why this is added outside of the loop. Also - consider having an "if" inside a loop.